File: go/merge_sort/merge_sort_ints_test.go
package merge_sort

import (
	"github.com/stretchr/testify/assert"
	"reflect"
	"testing"
)

func TestIntRandomCases(t *testing.T) {
	for range 1_000 {
		td := newRandomIntTestData()

		actual := append([]int{}, td.array...)
		MergeSortInts(actual)

		if !reflect.DeepEqual(td.expectedArray, actual) {
			assert.FailNowf(t, "",
				"MergeSortInts(%v):\n  Actual: %v\nExpected: %v",
				td.array, actual, td.expectedArray,
			)
		}
	}

}

File: python/merge_sort/merge_sort_ints_test.py
from merge_sort_ints import merge_sort_ints
from merge_sort_ints_test_data import RandomIntsTestData


def test_int_random_cases():
	for _ in range(1_000):
		td = RandomIntsTestData()

		got = td.array.copy()
		merge_sort_ints(got)

		assert got == td.want_array, \
			f'merge_sort_ints({td.array}):\nwant: {td.want_array}\n got: {got}'
